src/appengine/handlers/cron/fuzzer_weights.py

```diff
@@ -221,37 +214,31 @@ def _coverage_formatter(query_format, dataset):
     STARTUP_CRASH_SPECIFICATION,
 ]
 
-# Special specification used to modify previously altered weights to their
-# default values when they no longer match any other specifications.
-RESTORE_DEFAULT_SPECIFICATION = QuerySpecification(
-    adjusted_weight=1.0,
-    threshold=None,
-    query_format=None,
-    formatter=None,
-    reason='no longer matches any weight adjustment specifications')
+RESTORE_DEFAULT_MATCH = SpecificationMatch(
+    new_weight=1.0, reason='no longer matches any weight adjustment rules')
 
 
 def _query_helper(client, query):
   """Helper function to get fuzzer stats."""
   return client.query(query=query).rows
 
 
-def _update_match(matched_specifications, fuzzer, job, specification):
+def _update_match(matches, fuzzer, job, match):
   """Update the weight for a fuzzer/job."""
   key = (fuzzer, job)
-  old_match = matched_specifications.get(key, RESTORE_DEFAULT_SPECIFICATION)
+  old_match = matches.get(key, RESTORE_DEFAULT_MATCH)
 
-  new_weight = specification.adjusted_weight
-  old_weight = old_match.adjusted_weight
+  new_weight = match.new_weight
+  old_weight = old_match.new_weight
 
   # Always update the weight if the previous value is the default. This is
   # required to deal with specifications that are meant to set the weight above
   # 1.0. Otherwise, prioritize only the most penalizing match for this pairing.
-  if old_match == RESTORE_DEFAULT_SPECIFICATION or new_weight < old_weight:
-    matched_specifications[key] = specification
+  if old_match == RESTORE_DEFAULT_MATCH or new_weight < old_weight:
```

Comment:
How does this work for new fuzzers where new weight is 5.0 and is > default.

Reply:
If we don't have an entry yet we'll match the first part of this condition thanks to the default on 229 above (and we have another case where we might have assigned RESTORE_DEFAULT_MATCH on 289). That said, there are probably significantly more cases where we'll override it now. The crash rule in particular needs some kind of minimum threshold to avoid this.

Reply:
Added minimums.